enzyme/Enzyme/Enzyme.cpp

```diff
@@ -82,13 +86,17 @@ void HandleAutoDiff(CallInst *CI, TargetLibraryInfo &TLI, AAResults &AA) {
   unsigned truei = 0;
   IRBuilder<> Builder(CI);
 
+  bool AtomicAdd = llvm::Triple(CI->getParent()->getParent()->getParent()->getTargetTriple()).getArch() == Triple::nvptx ||
```

Comment:
Can't you look at the TLI instead?

Reply:
TLI didn't seem to have a method to extract the triple (only via construction)